.github/workflows/ci.yaml

```diff
@@ -0,0 +1,81 @@
+name: CI
+
+on:
+  pull_request:
+  push:
+    branches:
+    - main
+    - track/**
+
+# kill all previous jobs in the same branch if new commits are pushed
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  quality-checks:
+    name: Quality Checks
+    strategy:
+      matrix:
```

Comment:
I believe the matrix can be generated from the repository. If we automate this part, we could get rid of future issues with updating this list if any directory changes name or is removed.

Reply:
I thought about this, but you can see that two matrices are different. That is because not all the charms have itests

one possible step for us could be create itest tox environments for all the charms, even if it is a dummy one and just always passes

Reply:
Okay, let's keep this and improve if needed.